mat_dp_pipeline/data_sources/mat_dp_db.py

```diff
@@ -77,16 +78,37 @@ class MatDPDBIntensitiesSource(
 ):
     _materials_spreadsheet: Path
 
-    def __init__(self, materials_spreadsheet: Path):
-        self._materials_spreadsheet = materials_spreadsheet
+    def __init__(self, intensities: pd.DataFrame):
+        self._intensities = intensities
+
+    @classmethod
+    def from_excel(
+        cls,
+        spreadsheet: str | Path,
+        sheet_name: str = "Material intensities",
+        engine: Literal["xlrd", "openpyxl", "odf", "pyxlsb"] | None = None,
+        header=1,
+    ):
+        source = pd.read_excel(
+            Path(spreadsheet), sheet_name=sheet_name, header=header, engine=engine
+        )
+        return cls(
+            source,
+        )
+
+    @classmethod
+    def from_csv(
+        cls,
+        csv: str | Path,
+        sep: Optional[str] = None,
+        header=1,
+    ):
+        source = pd.read_csv(csv, sep=sep, header=header)
+        return cls(source)
 
     def _raw(self) -> pd.DataFrame:
         df = (
-            pd.read_excel(
-                self._materials_spreadsheet,
-                sheet_name="Material intensities",
-                header=1,
-            )
+            self._intensities.copy()
```

Comment:
`copy` isn't necessary here.

Reply:
I think there is? Otherwise we mess about with it on call, and call can be called multiple times in theory

Reply:
It's not in-place. If it was, it wouldn't return. Fine to leave if in doubt, can't say there's a lot of overhead here with a copy...